tests/build-rss.test.js

```diff
@@ -0,0 +1,101 @@
+const fs = require('fs');
+const path = require('path');
+const rssFeed = require('../scripts/build-rss');
+const { mockRssData, title, type, desc, outputPath } = require('./fixtures/rssData');
+
+jest.mock('../config/posts.json', () => mockRssData, { virtual: true });
+
+describe('rssFeed', () => {
+  const testOutputDir = path.join(__dirname, '..', 'public', 'test-output');
+
+  beforeEach(() => {
+    fs.mkdirSync(testOutputDir, { recursive: true });
+  });
+
+  afterEach(() => {
+    if (fs.existsSync(testOutputDir)) {
+      fs.readdirSync(testOutputDir).forEach(file => {
+        fs.unlinkSync(path.join(testOutputDir, file));
+      });
+      fs.rmdirSync(testOutputDir);
+    }
+  });
+
+  it('should generate RSS feed and write to file', () => {
+    try {
+      rssFeed(type, title, desc, outputPath);
+    } catch (err) {
+      console.error('Error encountered during test execution:', err)
```

Comment:
Instead of using `console.error`, store the `err` object inside a variable and during assertions, check that error object should be empty. This leads to the right approach to test the functions if they are executed  perfectly or not.